src/types/Recipient.ts

```diff
@@ -32,12 +32,27 @@ export class IntegrationRecipient extends Recipient {
     }
 }
 
-export class NotificationRecipient extends Recipient {
-    readonly sendToAdmin: boolean;
+export abstract class BaseNotificationRecipient extends Recipient {
```

Comment:
I'm not sure I see the reason for using an abstract class.... are you using it to prevent instantiation of BaseNotificationRecipient?

Reply:
yes - plus I don't need to implement `equals` (defined int the base). I think I could also get rid of `key` and generate it on the child classes too instead of receiving it.

Reply:
Couldn't those reasons be accomplished by using regular inheritance? Generally, I tend to see abstract for instances where you declare a class function then later define it under the inheritors (abstract functions) or for some sort of override, but here it seems to me that it's being used for normal inheritance.